frontend/src/app/user/user-settings/user-settings.component.ts

```diff
@@ -132,4 +130,22 @@ export class UserSettingsComponent implements OnInit {
         const userSettings = this.form.getRawValue();
         this.authService.newUserSettings(userSettings);
     }
+
+    private onSuccess(user: UserResponse) {
+        this.toastService.show({
+            header: "Settings updated",
+            body: "Your settings have been successfully updated.",
+            type: "success",
+        });
+        this.updateCache(user.id);
+    }
+
+    private updateCache(id: number) {
+        const cache = this.apollo.client.cache;
+        const identified = cache.identify({ __typename: 'UserType', id });
+        cache.evict({ id: identified, fieldName: 'firstName' });
+        cache.evict({ id: identified, fieldName: 'lastName' });
+        cache.evict({ id: identified, fieldName: 'fullName' });
```

Comment:
Is there any reason why you specifically want to evict these fields and not do `cache.evict({ id: identified })` in general?

Reply:
It's more precise because those are the only fields that are edited here. But I think those are also the only fields, so this is not really necessary. Setting a `fieldName` is more useful with the data entry forms, where evicting specific fields really avoids a lot of needless requests.